### PR TITLE
[dockerize]

### PR DESCRIPTION
**Pull request by siva2204**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      dockerfile: ./Dockerfile
+    restart: always
+    depends_on:
+      - db
+      - redis
+    ports:
+      - '${API_LOCAL_PORT}:${API_DOCKER_PORT}'
+
+volumes:
+  dauth_db:
```
Comment (anirudhvs) [inline docker-compose.yml:44]: Can we volume mount to pwd instead of using docker volumes?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# syntax=docker/dockerfile:1
```
Comment (anirudhvs) [inline Dockerfile:1]: This line can be removed right?

File: .env.example [hunk 3]
```diff
@@ -10,6 +10,15 @@ SMTP_PORT=
 SMTP_USER=
 SMTP_PASS=
 
+REDIS_HOST=redis
+REDIS_PORT=6379
+
+API_LOCAL_PORT=
+API_DOCKER_PORT=
+
+PMA_LOCAL_PORT=
```
Comment (anirudhvs) [inline .env.example:17]: Can you set this to 80 inside the compose file?

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+      - ./.docker.env
+    ports:
+      - '${PMA_LOCAL_PORT}:80'
+    depends_on:
+      - db
+
+  redis:
+    image: redis:alpine
+    container_name: dauth_redis
+    restart: always
```
Comment (anirudhvs) [inline docker-compose.yml:29]: Can we volume mount redis also? We might loose session data if container restarts.

File: docker-entry.sh [hunk 5]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+
+# migration
+prisma migrate dev
```
Comment (anirudhvs) [inline docker-entry.sh:4]: Migration will run on every container start which is unnecessary right? Unless we change some code. Can you make it run only on first start?

**Review by anirudhvs — APPROVED** [review]

**Review by anirudhvs — APPROVED** [review]

**Review by nishihere19 — APPROVED** [review]

**Pull request merged**